### PR TITLE
Optimizing variable initialization

### PR DESCRIPTION
**Pull request by lucming**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by lucming** [comment]
@josephburnett @mwielgus please help me review the changes,thanks :)

File: pkg/controller/podautoscaler/horizontal.go [hunk 1]
```diff
@@ -615,20 +615,13 @@ func (a *HorizontalController) reconcileAutoscaler(ctx context.Context, hpaShare
 		metricName            string
 	)
 
-	desiredReplicas := int32(0)
 	rescaleReason := ""
-
-	var minReplicas int32
-
+	var desiredReplicas, minReplicas int32 = 0, 1
```
Comment (pbetkier) [inline pkg/controller/podautoscaler/horizontal.go:619]: IMO `var minReplicas int32 = 1` right before the `hpa.Spec.MinReplicas != nil` if makes the code more readable 👍

IMO multiple variable initialization makes the code less readable. Please add `var desiredReplicas int32` right before `scale.Spec.Replicas == 0 && minReplicas != 0` if instead.

File: staging/src/k8s.io/component-helpers/apps/poddisruptionbudget/helpers.go [hunk 2]
```diff
@@ -28,21 +28,21 @@ func UpdateDisruptionAllowedCondition(pdb *policy.PodDisruptionBudget) {
...
-			Type:               policy.DisruptionAllowedCondition,
-			Reason:             policy.InsufficientPodsReason,
-			Status:             metav1.ConditionFalse,
-			ObservedGeneration: pdb.Status.ObservedGeneration,
-		})
+		condition.Reason = policy.InsufficientPodsReason
+		condition.Status = metav1.ConditionFalse
 	}
+
+	apimeta.SetStatusCondition(&pdb.Status.Conditions, condition)
```
Comment (pbetkier) [inline staging/src/k8s.io/component-helpers/apps/poddisruptionbudget/helpers.go:45]: It's arguable if the refactoring makes the code more readable. IMO the scattered condition config makes it less clear what is set. Please revert.

**Review by pbetkier — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/component-helpers/apps/poddisruptionbudget/helpers.go (hunk 2 above)
Comment (lucming) [reply staging/src/k8s.io/component-helpers/apps/poddisruptionbudget/helpers.go]: > It's arguable if the refactoring makes the code more readable. IMO the scattered condition config makes it less clear what is set. Please revert.
already rolled back.

File: pkg/controller/podautoscaler/horizontal.go (hunk 1 above)
Comment (lucming) [reply pkg/controller/podautoscaler/horizontal.go]: > IMO `var minReplicas int32 = 1` right before the `hpa.Spec.MinReplicas != nil` if makes the code more readable 👍
> 
> IMO multiple variable initialization makes the code less readable. Please add `var desiredReplicas int32` right before `scale.Spec.Replicas == 0 && minReplicas != 0` if instead.

done.

**Review by pbetkier — APPROVED** [review]
/lgtm

**Pull request closed without merging**